### ISSUE TITLE
Warnings are regarded as uncommitted files and prevent my gem from being released.

### ISSUE THREAD

**Issue by khiav223577** [bundler/bundler#6919]
I got an error when trying to release my gem.
```bash
$ rake release
rake aborted!
There are files that need to be committed first.
```

I traced and found that the error message came from here:
https://github.com/bundler/bundler/blob/88ea68f800d463f09622b9ef6c4331f704a6bd98/lib/bundler/gem_helper.rb#L148

It seems that the warnings from `git diff` command will be regarded as some files are changed and are not committed.

```bash
$ git diff --exit-code
warning: LF will be replaced by CRLF in .travis.yml.
The file will have its original line endings in your working directory
```

I have no idea how to get rid of this.
I think it is a kind of bug of `bundler` which regard warnings as uncommitted files.

```bash
$ git add .
$ git diff --exit-code
$ rake release
rake aborted!
There are files that need to be committed first.
$ git diff --exit-code
warning: LF will be replaced by CRLF in .travis.yml.
The file will have its original line endings in your working directory
```

**Comment by hahmed** [comment]
I can take this issue.

**Comment by hahmed** [comment]
@khiav223577 I cannot seem to replicate your issue.

I tried this with a repo of my own, here is an example of my code:

```
myrepo blocked-users-in-export % git diff --exit-code
diff --git a/spec/myrepo/projects/parse_user_data_spec.rb b/spec/myrepo/projects/parse_user_data_spec.rb
index a5c12a7f5..42f403786 100644
--- a/spec/myrepo/projects/parse_user_data_spec.rb
+++ b/spec/myrepo/projects/parse_user_data_spec.rb
@@ -2,7 +2,7 @@ require 'spec_helper'
 
 describe Projects::ParseUserData do
   ATTRIBUTES = %w(
-    id date_of_birth blocked_at
+    id date_of_birth 
   ).freeze
 
   let(:user_attributes) { attributes_for(:user, :for_export) }
```

Once I commit the files e.g.

`myrepo blocked-users-in-export % git commit -am 'fixed test'`

I re-ran the diff command that bundler runs again

`myrepo blocked-users-in-export % git diff --exit-code`  

The output from that was nothing, looking at your issue, I can see

```
$ git add .
$ git diff --exit-code
```

Can you commit the files too, then re-run and see what happens?

If I have this totally wrong, please let me know so I can replicate - thanks!

**Comment by khiav223577** [comment]
You could clone [this repo](https://github.com/khiav223577/atomically) and test with it.
The following is the snapshot of the problem I encountered.

![image](https://user-images.githubusercontent.com/4011729/53044390-cc29bd80-34c5-11e9-8c77-22334f69dc17.png)

**Comment by hahmed** [comment]
We can ignore the warning regarding line endings, that will not stop you releasing your gem.

You have windows machine, that will be a little tricky for me to debug - as I don't have a windows machine at hand unfortunately. Let's see if we can work through this anyhow;

Thanks for showing your console output, that really helps

`git diff --exit-code` - looks like it's not retuning an error

`git diff-index --quiet --cached HEAD` - does this return an error?

If neither of the above return errors, it's likely the issue you raised is related to how the shell script is run in windows and possibly the return code from the shell script.

**Comment by hahmed** [comment]
You may need to edit your gitconfig and add something like `git config --global difftool.trustExitCode true` to get this to work.

**Comment by khiav223577** [comment]
only `git diff --exit-code` is not success

![image](https://user-images.githubusercontent.com/4011729/53049882-0188d800-34d3-11e9-934d-f3d58882126e.png)

**Comment by samyak-jn** [comment]
There might be a possibility for you to do configurations for your git, as only your 
git diff -- exit-code, hence configuring it may help.
           `$ git config --global`
           `$ git config --global core.autocrlf true `
or,       `**$ git config --global core.safecrlf false** `
Warnings can be ignored, I think so! I was having somewhat different errors for warnings but they can be ignored.
Git looks is the ~/.gitconfig (or ~/.config/git/config) file, which is specific to each user. You can make Git read and write to this file by passing the --global option.
Just try these command as suitable to your distros.
    For turning the warning off, after searching crlf error 

> can be removed using (core.autocrlf/core.safecrlf)

`$ git config --global core.safecrlf false` to disable warning and keep it functioning.    

If you’re a Windows programmer doing a Windows-only project, then you can turn off this functionality, recording the carriage returns in the repository by setting the config value to false:

`$ git config --global core.autocrlf false`
If I'm going somewhere wrong, please correct me so I can try help in a different way.

**Comment by khiav223577** [comment]
I found that the configurations for my git is different between in `bash` and in `irb`.
In `bash`, `git config --global core.autocrlf` prints `true`.
In `irb`, `%x[git config --global core.autocrlf]` prints `""`.

And I finally resolved this by doing configurations for the git in `irb`. 😄 
```rb
`git config --global core.autocrlf true`
```

No idea why the configurations are not the same in `irb`.
